src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -77,479 +47,2365 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    static s32 sFlameSpawned = false;
+
+    if (!sFlameSpawned) {
+        CsCmdActorAction* npcAction = EnXc_GetCsCmd(globalCtx, 0);
+        f32 xPos = npcAction->startPos.x;
+        f32 yPos = npcAction->startPos.y;
+        f32 zPos = npcAction->startPos.z;
+        this->flameActor = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_LIGHT, xPos, yPos, zPos, 0, 0, 0, 5);
+        sFlameSpawned = true;
+    }
+}
+
+void EnXc_SetupFlamePos(EnXc* this, GlobalContext* globalCtx) {
+    Vec3f* attachedPos;
+    CsCmdActorAction* npcAction = EnXc_GetCsCmd(globalCtx, 0);
+
+    if (this->flameActor != NULL) {
+        attachedPos = &this->flameActor->world.pos;
+        if (!this) {}
+        attachedPos->x = npcAction->startPos.x;
+        attachedPos->y = npcAction->startPos.y;
+        attachedPos->z = npcAction->startPos.z;
+    }
+}
+
+void EnXc_DestroyFlame(EnXc* this) {
+    if (this->flameActor != NULL) {
+        Actor_Kill(this->flameActor);
+        this->flameActor = NULL;
+    }
+    Actor_Kill(&this->actor);
+}
+
+void EnXc_InitFlame(EnXc* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 sceneNum = globalCtx->sceneNum;
+    static s32 D_80B41DA8 = 1;
+
+    if (sceneNum == SCENE_SPOT17) {
+        CsCmdActorAction* npcAction = EnXc_GetCsCmd(globalCtx, 0);
+        if (npcAction != NULL) {
+            s32 action = npcAction->action;
+            if (D_80B41DA8 != action) {
+                if (action != 1) {
+                    EnXc_SpawnFlame(this, globalCtx);
+                }
+
+                if (action == 1) {
+                    EnXc_DestroyFlame(this);
+                }
+
+                D_80B41DA8 = action;
+            }
+
+            EnXc_SetupFlamePos(this, globalCtx);
+        }
+    }
+}
+
+void func_80B3D48C(EnXc* this, GlobalContext* globalCtx) {
+    CutsceneContext* csCtx = &globalCtx->csCtx;
+    CsCmdActorAction* linkAction = csCtx->linkAction;
+    s16 yaw;
+
+    if (linkAction != NULL) {
+        yaw = (u16)linkAction->rot.y + 0x8000;
+    } else {
+        Player* player = PLAYER;
+        yaw = player->actor.world.rot.y + 0x8000;
+    }
+
+    this->actor.world.rot.y = yaw;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+AnimationHeader* EnXc_GetCurrentHarpAnim(GlobalContext* globalCtx, s32 index) {
+    AnimationHeader* animation = &gShiekPlayingHarp5Anim;
+    CsCmdActorAction* npcAction = EnXc_GetCsCmd(globalCtx, index);
+
+    if (npcAction != NULL) {
+        u16 action = npcAction->action;
+        if (action == 11) {
+            animation = &gShiekPlayingHarp3Anim;
+        } else if (action == 12) {
+            animation = &gShiekPlayingHarp2Anim;
+        } else if (action == 13) {
+            animation = &gShiekPlayingHarp4Anim;
+        } else {
+            animation = &gShiekPlayingHarp5Anim;
+            if (action == 23) {
+                animation = &gShiekPlayingHarpAnim;
+            }
+        }
```

Comment:
This might match and seems better?
```suggestion
        } else if (action == 23) {
            animation = &gShiekPlayingHarpAnim;
        } else {
            animation = &gShiekPlayingHarp5Anim;
        }
```